ic-stable-structures/src/memory_mapped_files/memory.rs

```diff
@@ -12,56 +15,87 @@ const WASM_PAGE_SIZE_IN_BYTES: u64 = 65536;
 pub struct MemoryMappedFileMemoryManager {
     base_path: PathBuf,
     is_persistent: bool,
+    created_memory_resources: RwLock<BTreeMap<PathBuf, MemoryMappedFileMemory>>,
 }
 
 impl MemoryMappedFileMemoryManager {
     pub fn new(base_path: PathBuf, is_persistent: bool) -> Self {
         Self {
             base_path,
             is_persistent,
+            created_memory_resources: Default::default(),
+        }
+    }
+
+    pub fn flush_and_save_copies_to(&self, path: impl AsRef<Path>) -> Result<(), MemMapError> {
+        let created_memory_resources = self.created_memory_resources.read();
+        for (file_path, memory) in created_memory_resources.iter() {
+            let file_name = file_path
+                .file_name()
+                .ok_or(MemMapError::InvalidSourceFileName)?;
+            let new_path = path.as_ref().join(
+                file_name
+                    .to_str()
+                    .ok_or(MemMapError::InvalidSourceFileName)?,
+            );
+
+            memory.save_copy(new_path)?;
+        }
+
+        Ok(())
+    }
```

Comment:
where is the `flush` part performed?

Reply:
It happens inside `MemoryMappedFile::save_copy`

Reply:
So, two things:
1. Why do you have to explicitly state it in the name of the function?
2. I guess we need a dedicated "flush_all" function